### PR TITLE
feat: add uniswap ui

### LINKED ISSUE #18: Add Uniswap UI

### PR DESCRIPTION
**Pull request by Bender-22**
closes #18

### CODE REVIEW

File: abis/UniswapPoolRewards.json [hunk 1]
```diff
@@ -0,0 +1,396 @@
+{
```
Comment (nya-elimuai) [inline abis/UniswapPoolRewards.json:1]: Is this mainnet or Rinkeby?

File: pages/uniswap.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
+import Head from 'next/head'
+import Footer from '../components/Footer'
+import Header from '../components/Header'
+import { Alert } from '@mui/material'
+import { WagmiConfig, createClient, configureChains, defaultChains, useContractRead, usePrepareContractWrite, useContractWrite } from 'wagmi'
+import { publicProvider } from 'wagmi/providers/public'
+import UniswapV2Pair from '../abis/UniswapV2Pair.json'
+import UniswapPoolRewards from '../abis/UniswapPoolRewards.json'
```
Comment (nya-elimuai) [inline pages/uniswap.tsx:8]: We need to change the way ABIs are loaded when moving from Rinkeby to Mainnet.

File: pages/uniswap.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+function ApproveButton() {
+  console.log('ApproveButton')
+
+  const { config, error } = usePrepareContractWrite({
+    addressOrName: '0x9936bdcd16e8c709c4cb8d7b871f0011b4cc65de',
+    contractInterface: UniswapV2Pair.abi,
+    functionName: 'approve',
+    args: ['0xb8b6430b58a4fbc57bd5cd7715d318065f2dcd81', ethers.utils.parseUnits('1000')]
```
Comment (nya-elimuai) [inline pages/uniswap.tsx:38]: Replace hardcoded values with variables.

File: pages/uniswap.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+function DepositButton({ depositAmount }: any) {
+  console.log('DepositButton')
+
+  console.log('depositAmount:', depositAmount)
+
+  const { config, error } = usePrepareContractWrite({
+    addressOrName: '0xb8b6430b58a4fbc57bd5cd7715d318065f2dcd81',
+    contractInterface: UniswapPoolRewards.abi,
```
Comment (nya-elimuai) [inline pages/uniswap.tsx:91]: Replace hardcoded values with variables.

File: pages/uniswap.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                      console.log('amount:', amount)
+                      let depositButtonElement = (document.getElementById('depositButton') as HTMLButtonElement)
+                      // if (amount > 0) {
+                      //   depositButtonElement.disabled = false
+                      // } else {
+                      //   depositButtonElement.disabled = true
+                      // }
+                    }}
+                    />
+                <DepositButtonWrapper depositAmount={0.1} />
```
Comment (nya-elimuai) [inline pages/uniswap.tsx:180]: Replace hardcoded amount with variable.

**Review by nya-elimuai — CHANGES_REQUESTED** [review]
Requested to replace hardcoded values.

**Review by jo-elimu — CHANGES_REQUESTED** [review]
Instead of using 1,000, can we fetch the account's amount of pool tokens, and then ask for the approval of that specific amount?

**Review by jo-elimu — CHANGES_REQUESTED** [review]
I'm getting a `ContractFunctionExecutionError` in the UI: https://deploy-preview-21--web3-liquidity-rewards-ui.netlify.app/uniswap

Screenshot 2023-06-22 171432

**Review by jo-elimu — APPROVED** [review]

**Review by nya-elimu — APPROVED** [review]

**Pull request merged**